src/Network/WebSockets/TSSocketManager.m

```diff
@@ -127,13 +146,28 @@ - (void)becomeActive
 }
 
 + (void)resignActivity {
-    SRWebSocket *socket = [[self sharedManager] websocket];
-    [socket close];
+    OWSAssert([NSThread isMainThread]);
```

Comment:
This blows up in AppDelegate:

```
- (void)applicationWillResignActive:(UIApplication *)application {
       // ...Snip for brevity

    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
      if ([TSAccountManager isRegistered]) {
          dispatch_sync(dispatch_get_main_queue(), ^{
            [self protectScreen];
            [[[Environment getCurrent] signalsViewController] updateInboxCountLabel];
          });
          [TSSocketManager resignActivity]; // !!! <-- I suppose this could move up a line into the "main" block.
      }

       // ...Snip for brevity
    });
}
```